common/ast/src/ast/expr.rs

```diff
@@ -59,6 +59,20 @@ pub enum Expr<'a> {
         subquery: Box<Query<'a>>,
         not: bool,
     },
+    /// `expr op ANY (SELECT ...)`
+    AnySubquery {
+        span: &'a [Token<'a>],
+        expr: Box<Expr<'a>>,
+        subquery: Box<Query<'a>>,
+        op: ComparisonOperator,
+    },
+    /// `expr op ALL (SELECT ...)`
+    AllSubquery {
+        span: &'a [Token<'a>],
+        expr: Box<Expr<'a>>,
+        subquery: Box<Query<'a>>,
+        op: ComparisonOperator,
+    },
```

Comment:
I think it's better to use a tag `SubqueryModifier` to represent the modifier `ALL` and `ANY`.

By the way, we can support `SOME` modifier as well, which is essentially the same with `ANY`.

Reply:
> I think it's better to use a tag `SubqueryModifier` to represent the modifier `ALL` and `ANY`

Now I treat ANY and ALL as separate operators, to avoid the conflict between `ComparisonOperator` and `BinaryOperator`